### PR TITLE
8307855: update for deprecated sprintf for src/utils

### PR DESCRIPTION
**Pull request by XueleiFan**
Hi,

May I have this update reviewed?

The sprintf is deprecated in Xcode 14, and Microsoft Virtual Studio, because of security concerns. The issue was addressed in [JDK-8296812](https://bugs.openjdk.org/browse/JDK-8296812) for building failure, and [JDK-8299378](https://bugs.openjdk.org/browse/JDK-8299378)/[JDK-8299635](https://bugs.openjdk.org/browse/JDK-8299635)/[JDK-8301132](https://bugs.openjdk.org/browse/JDK-8301132) for testing issues . This is a break-down update for sprintf uses in the src/utils directory.

Thanks,
Xuelei

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8307855](https://bugs.openjdk.org/browse/JDK-8307855): update for deprecated sprintf for src/utils

### Reviewers
 * [Tobias Hartmann](https://openjdk.org/census#thartmann) (@TobiHartmann - **Reviewer**) ⚠️ Review applies to [6a7d4e69](https://git.openjdk.org/jdk/pull/13915/files/6a7d4e69ef30d9287382d2cbb1990186190d5725)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13915/head:pull/13915` \
`$ git checkout pull/13915`

Update a local copy of the PR: \
`$ git checkout pull/13915` \
`$ git pull https://git.openjdk.org/jdk.git pull/13915/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13915`

View PR using the GUI difftool: \
`$ git pr show -t 13915`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13915.diff">https://git.openjdk.org/jdk/pull/13915.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13915#issuecomment-1542796362)

### CODE REVIEW

File: src/utils/hsdis/binutils/hsdis-binutils.c
Comment (TobiHartmann) [inline src/utils/hsdis/binutils/hsdis-binutils.c:2]:
```suggestion
 * Copyright (c) 2008, 2023, Oracle and/or its affiliates. All rights reserved.
```

**Review by TobiHartmann — APPROVED** [review]
Looks good to me otherwise.

File: src/utils/hsdis/binutils/hsdis-binutils.c
Comment (XueleiFan) [reply src/utils/hsdis/binutils/hsdis-binutils.c]: Oops.  Thank you for the catch!

File: src/utils/hsdis/binutils/hsdis-binutils.c
Comment (kimbarrett) [inline src/utils/hsdis/binutils/hsdis-binutils.c:222]: This shouldn't be a macro.  Among other reasons, see the style guide.

File: src/utils/hsdis/binutils/hsdis-binutils.c
Comment (kimbarrett) [inline src/utils/hsdis/binutils/hsdis-binutils.c:251]: What is the value for p that is used in the call to remaining_buflen?
It is being assumed to be the value after the assignment by the first
argument. However, according to the standard, it is unspecified, and
the whole snprintf call invokes UB. This is because there aren't any
sequence points between the update of p in the first argument and that
reference. (C++17 changes this, but we aren't using C++17 yet.)

**Review by kimbarrett — COMMENTED** [review]
A couple of issues noticed after this PR has been integrated.

**Comment by TobiHartmann** [comment]
The change is also missing a second review. I'm backing it out for now with https://github.com/openjdk/jdk/pull/13975. Let's redo with [JDK-8308071](https://bugs.openjdk.org/browse/JDK-8308071).

Thanks,
Tobias

File: src/utils/hsdis/binutils/hsdis-binutils.c
Comment (XueleiFan) [reply src/utils/hsdis/binutils/hsdis-binutils.c:222]: May I have an reference to the style guide?

**Comment by XueleiFan** [comment]
> The change is also missing a second review. I'm backing it out for now with #13975. Let's redo with [JDK-8308071](https://bugs.openjdk.org/browse/JDK-8308071).
> 
> Thanks, Tobias

Thanks!

File: src/utils/hsdis/binutils/hsdis-binutils.c
Comment (XueleiFan) [reply src/utils/hsdis/binutils/hsdis-binutils.c:251]: For the 1st line (line 249), it is fine to use bufsize directly. The p in remaining_buflen is used to calculate the remaining length of the target buffer.  To know the remaining buffer length, I need to know the pointer to the beginning buffer (the 'buf' parameter), the pointer to the beginning of the unused buffer (the 'p' parameter), and the total size of the buffer (the 'bufsize' parameter).  The initial value of p is assigned to buf(p = buf), and then move forward by the used size ( p += strlen(p)).

It's a good point to me about the update sequence of p, and I will make an update in a new PR.

**Pull request closed without merging**
